app/src/main/java/org/oppia/android/app/onboarding/CreateProfileFragmentPresenter.kt

```diff
@@ -107,19 +142,102 @@ class CreateProfileFragmentPresenter @Inject constructor(
 
     binding.onboardingNavigationBack.setOnClickListener { activity.finish() }
     binding.createProfileEditPictureIcon.setOnClickListener {
-      activityResultLauncher.launch(
-        galleryIntent
-      )
+      activityResultLauncher.launch(galleryIntent)
     }
     binding.createProfilePicturePrompt.setOnClickListener {
-      activityResultLauncher.launch(
-        galleryIntent
-      )
+      activityResultLauncher.launch(galleryIntent)
     }
     binding.createProfileUserImageView.setOnClickListener {
-      activityResultLauncher.launch(
-        galleryIntent
-      )
+      activityResultLauncher.launch(galleryIntent)
     }
   }
+
+  private fun updateProfileDetails(profileName: String) {
+    profileManagementController.updateNewProfileDetails(
+      profileId = profileId,
+      profileType = profileType,
+      avatarImagePath = selectedImageUri,
+      colorRgb = selectUniqueRandomColor(),
+      newName = profileName,
+      isAdmin = true
+    ).toLiveData().observe(
+      fragment,
+      { result ->
+        when (result) {
+          is AsyncResult.Success -> {
+            createProfileViewModel.hasErrorMessage.set(false)
+
+            val params = IntroActivityParams.newBuilder()
+              .setProfileNickname(profileName)
+              .build()
+
+            val intent =
+              IntroActivity.createIntroActivity(activity).apply {
+                putProtoExtra(IntroActivity.PARAMS_KEY, params)
+                decorateWithUserProfileId(profileId)
+              }
+
+            fragment.startActivity(intent)
+          }
+          is AsyncResult.Failure -> {
+            createProfileViewModel.hasErrorMessage.set(true)
+
+            val errorMessage = when (result.error) {
+              is ProfileManagementController.ProfileNameOnlyLettersException ->
+                appLanguageResourceHandler.getStringInLocale(
+                  R.string.add_profile_error_name_only_letters
+                )
+              else -> result.error.localizedMessage
```

Comment:
Where does this come from? I'm a bit concerned about a human-readable message being sourced from the domain layer (vs. passing an enum value that can be switched to retrieve the correct translated string to display to the user).

Reply:
But can it still actually show up? Under what circumstances?

We use ``ProfileNameOnlyLettersException`` specifically so that we can retrieve a localized (and hopefully actionable) error message for the user. Why can't we introduce another or multiple exception types for other possibilities, as well?

Reply:
I created a new exception, `UnknownProfileTypeException, ` which I determined to be the likely scenario when ProfileType was not passed.

Reply:
I think this is much clearer--thanks!